cmake/kokkos_enable_options.cmake

```diff
@@ -54,6 +54,7 @@ endif()
 kokkos_enable_option(
   SYCL_RELOCATABLE_DEVICE_CODE ${SYCL_RDC_DEFAULT} "Whether to enable relocatable device code (RDC) for SYCL"
 )
+kokkos_enable_option(SYCL_OUT_OF_ORDER_QUEUES OFF "Whether to make Kokkos use out-of-order queues internally")
```

Comment:
Are they issues not tied to toolchains versions?
Please explain why we want/need to push that choice onto the user.